test/type-applications/Main.hs

```diff
@@ -331,9 +345,135 @@ propUnifyWildcardRigid = forAllShrink arbitrary shrink $ \(scope, index) ->
          in withRenamedVals (Identity argT) $ \(Identity argT') ->
               let lhs = ThunkT . Comp1 $ Abstraction (Wildcard 1 2 ix0) :--:> ReturnT integerT
                   expected = Left . DoesNotUnify lhs $ argT'
-                  actual = checkApp f [Just argT']
+                  actual = checkApp M.empty f [Just argT']
                in expected === actual
 
+-- Simple datatype unification unit test. Checks whether `data Unit = Unit` has the expected BB form
+testMonotypeBB :: TestTree
+testMonotypeBB = testCase "unitBbf" $ do
+  let expected = Comp1 $ Abstraction (BoundAt Z ix0) :--:> ReturnT (Abstraction $ BoundAt Z ix0)
+  expected' <- failLeft . runRenameM . renameCompT $ expected
+  actual <- case fromJust . (view #bbForm <=< M.lookup "Unit") $ tyAppTestDatatypes of
+    ThunkT inner -> either (throwIO . userError . show) pure . runRenameM $ renameCompT inner
+    _ -> assertFailure "BB form not a thunk!"
+  assertEqual "unit bbf" expected' actual
```

Comment:
I'm not sure this test belongs here. Instead, this should go in the BB form suite.